xarray/core/duck_array_ops.py

```diff
@@ -54,6 +58,59 @@
 dask_available = module_available("dask")
 
 
+HANDLED_EXTENSION_ARRAY_FUNCTIONS: dict[Callable, Callable] = {}
+
+
+def implements(numpy_function):
```

Comment:
This still feels quite unnecessary to me. Can we move these methods to `ExtensionDuckArray` and make `HANDLED_EXTENSION_ARRAY_FUNCTIONS` a hardcoded class property there.

Reply:
Ah I did not realize that, apologies.

In that case let's hold on this suggestion. However, could you move these functions in to `indexing.py`so the logic is all contained to one place.

We have a dev meeting next Wednesday (https://github.com/pydata/xarray/issues/4001) where we can discuss merging this. Can you attend?

Reply:
Yes!

Reply:
*Only first 30 minutes

Reply:
Also, done!

Reply:
It's only a 30 minute meeting so yay!